src/lib/y2partitioner/dialogs/main.rb

```diff
@@ -40,7 +43,13 @@ def run(system, initial)
         res = nil
         loop do
           res = super()
-          break if res != :redraw
+
+          next if res == :redraw
+          if res == :abort && Yast::Mode.installation
```

Comment:
also old one when pressing abort do not abort whole installation, just goes back ( verified manually on leap 42.1 ).